src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -93,6 +93,13 @@ private static boolean isValidPath(Path filePath) {
      */
     public void save(AddressBook addressBook) throws StorageOperationException {
 
+    	
+    	/*
+    	 * Test if xml file still exits before saving
+    	 */
+    	if(!path.toFile().exists()){
```

Comment:
This is the right place to add this new exception.

Note that you should also think about the follow up:
-  Who should handle this exception and how? 
- Note that the original code is rather robust as it always recreate the file, so you can consider whether we can still do that, but alert the user that something strange is going on, i.e. "Save file disappears/corrupted, but we have restored it for you!"